include/mbedtls/gcm.h

```diff
@@ -231,19 +231,36 @@ int mbedtls_gcm_auth_decrypt( mbedtls_gcm_context *ctx,
  * \param iv        The initialization vector. This must be a readable buffer of
  *                  at least \p iv_len Bytes.
  * \param iv_len    The length of the IV.
+ *
+ * \return          \c 0 on success.
+ */
+int mbedtls_gcm_starts( mbedtls_gcm_context *ctx,
+                        int mode,
+                        const unsigned char *iv,
+                        size_t iv_len );
+
+/**
+ * \brief           This function starts a GCM encryption or decryption
+ *                  operation.
+ *
+ * \note            This function may only be called once per operation:
+ *                  you must pass the whole associated data in a single
+ *                  call. This limitation will be lifted in a future version
+ *                  of Mbed TLS.
+ *
+ * \param ctx       The GCM context. This must have been started with
+ *                  mbedtls_gcm_starts() and must not have yet received
+ *                  any input with mbedtls_gcm_update().
```

Comment:
That's a “must” as in if you don't do this the behavior is undefined, and not a “must” as in if you don't do this the function returns a predictable error code. I think that adding extra state machine checks is out of scope (and the lack of check may be deliberate because this is a low-level module and checks cost code size, although my personal opinion is that size checks would be worth the few bytes of code but not worth the time coding them now).

Reply:
Yes I agree thanks for clarifying the context.